docs/local-development.md

```diff
@@ -0,0 +1,46 @@
+# Local Development
+
+This document aims to aid contributors for running this package in a local [k3d][k3d] cluster. If developing in a remote
+cluster, it should be as simple as configuring `KO_DOCKER_REPO` to point to a registry of your choosing, however when 
+running a k3d cluster in tandem with ko builds, the process is a little more convoluted.
+
+Pre-requisites:
+
+- [ko][ko]
+- [k3d][k3d]
+
+Create a cluster with a dedicated registry and traefik disabled (knative requires its own networking implementation 
+rather than using traefik - we use kourier for local dev) 
+
+```sh
+k3d cluster create knative \
+    --api-port 6550 \
+    --servers 1 \
+    --image rancher/k3s:v1.21.9-k3s1 \
+    --port 80:80@loadbalancer \
+    --port 443:443@loadbalancer \
+    --k3s-arg '--no-deploy=traefik@server:*' \
+    --registry-create k3d-ko.localhost:12345 \
+    --wait --verbose
+```
+
+Deploy Knative eventing to your cluster, following the getting started guide of your choosing from 
+[Installing Knative][install-knative].
+
+> Depending on your OS, you need to ensure `k3d-k0.localhost` resolves to `127.0.0.1`. If your OS does not do this by 
+> default, the simplest method is to add a line to `/etc/hosts`:
+> 
+> ```
+> 127.0.0.1 k3d-ko.local
+> ```
+
+Deploy eventing-natss to your cluster (ensuring your `kubectl` context is configured to your local cluster):
+
+```sh
+ko apply -f config/webhook
+ko apply -f config/jetstream
```

Comment:
And you may also need the `--insecure-registry` flag set:

```sh
KO_DOCKER_REPO=k3d-ko.local:12345 ko apply --insecure-registry -f config/webhook
KO_DOCKER_REPO=k3d-ko.local:12345 ko apply --insecure-registry -f config/jetstream
```

Reply:
Hi @dan-j ! I've tried out this PR for the basic pingsource example works, following the instructions in local-development.md by applying the files in webhook and jetstream, seeing some cloudevents in the logs on the event-display service

I'm currently testing this locally using k3s running on Rancher Desktop

The files I deployed from the example folder are config-nats.yaml, nats.yaml and ping-demo-default.yaml

Will be reading through and understanding more on this implementation over the next few days

Reply:
We would like to start developing with NATS JetStream and Knative. Is there anything my team can do help update/validate this PR, so it can be merged?